packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -757,37 +758,55 @@ function compileProofEntryConstraints(
   // Deal with equality comparision and POD ownership, which share circuitry.
   let firstOwnerIndex = undefined;
   const entryEqualToOtherEntryByIndex: bigint[] = [];
+  const entryIsEqualToOtherEntry: bigint[] = [];
   const virtualEntryEqualToOtherEntryByIndex: bigint[] = [];
+  const virtualEntryIsEqualToOtherEntry: bigint[] = [];
 
   for (const entryInfo of entryMap.values()) {
+    const equalToOtherEntryByIndex = isVirtualEntryName(entryInfo.entryName)
+      ? virtualEntryEqualToOtherEntryByIndex
+      : entryEqualToOtherEntryByIndex;
+    const isEqualToOtherEntry = isVirtualEntryName(entryInfo.entryName)
+      ? virtualEntryIsEqualToOtherEntry
+      : entryIsEqualToOtherEntry;
+
     // An entry is always compared either to the first owner entry (to ensure
     // only one owner), or to another entry specified by config, or to itself
     // in order to make the constraint a nop.
     if (entryInfo.entryConfig.isOwnerID) {
       if (firstOwnerIndex === undefined) {
         firstOwnerIndex = entryInfo.entryIndex;
-      } else if (entryInfo.entryConfig.equalsEntry !== undefined) {
+      } else if (
```

Comment:
I honestly can't remember why this is an `else if`.  It's not valid to combine isOwnerID with equalsEntry anytime, including on the first one.  But that's checked elsewhere.  I'd normally only do this kind of a double-check if it were necessary to make the next line valid (e.g. checking something is defined to avoid a compiler warning).
I'd be okay with leaving this clause as-is, or removing it.